elf_ctx.c

```diff
@@ -83,23 +83,24 @@
 #define AT_SYSINFO_EHDR		33
 #define AT_SYSINFO		32
 
+#define AUXV_VALUES_PER_ENTRY 	2
+
+struct aux_vector_entry {
+	long key;
+	long val;
+};
+
 #if CONFIG_ARCH_X86_64
 	static const char *auxv_platform = "x86_64";
 #else
 #error "Unsupported architecture"
 #endif /* CONFIG_ARCH_X86_64 */
 
-#define push_auxv(ctx, key, val)				\
-	do {							\
-		ukarch_rctx_stackpush(ctx, (long) val);		\
-		ukarch_rctx_stackpush(ctx, (long) key);		\
-	} while (0)
-
 void elf_ctx_init(struct ukarch_ctx *ctx, struct elf_prog *prog,
 		  int argc, char *argv[], char *environ[],
 		  uint64_t rand[2])
 {
-	int i, envc;
+	int i, envc, stackvalc, stackval_len;
```

Comment:
Maybe you want to cal it `elfvec_len` for `ELF vectors length`? The count may no be needed if you update the computation formular.